lib/util/CopyPasteUtil.js

```diff
@@ -1,43 +1,37 @@
 import {
-  forEach
+  filter,
+  find
 } from 'min-dash';
 
-
-export function getTopLevel(elements) {
-  var topLevel = {},
-      parents = [],
-      result = [],
-      clearedParents = [];
-
-  forEach(elements, function(element) {
-    var parent = element.parent;
-
-    if (!topLevel[parent.id]) {
-      topLevel[parent.id] = [];
-    }
-
-    if (parents.indexOf(parent.id) === -1) {
-      parents.push(parent.id);
-    }
-
-    topLevel[parent.id].push(element);
-  });
-
-  forEach(parents, function(parent) {
-    forEach(topLevel[parent], function(element) {
-      if (topLevel[element.id]) {
-        clearedParents.push(element.id);
-      }
+/**
+ * Get parent elements.
+ *
+ * @param {Array<djs.model.base>} elements
+ *
+ * @returns {Array<djs.model.Base>}
+ */
+export function getParents(elements) {
+
+  // find elements that are not children of any other elements
+  return filter(elements, function(element) {
```

Comment:
This implementation is super slow as it will run `elements.length * elements.length` times.

Does it have to be fast? If yes, let's look into :speedboat: it up.

Reply:
Okay, that might be an issue, let's look into it. 💪